rpc_util_test.go

```diff
@@ -290,3 +293,138 @@ func BenchmarkGZIPCompressor512KiB(b *testing.B) {
 func BenchmarkGZIPCompressor1MiB(b *testing.B) {
 	bmCompressor(b, 1024*1024, NewGZIPCompressor())
 }
+
+// compressData compresses data using gzip and returns the compressed bytes.
+func compressData(data []byte) []byte {
+	var buf bytes.Buffer
+	gz := gzip.NewWriter(&buf)
+	_, _ = gz.Write(data)
+	_ = gz.Close()
+	return buf.Bytes()
+}
+
+// TestDecompress tests the decompress function with various scenarios, including
+// successful decompression, error handling, and edge cases like overflow or
+// premature data end. It ensures that the function behaves correctly with different
+// inputs, buffer sizes, and error conditions, using the "gzip" compressor for testing.
+func TestDecompress(t *testing.T) {
+	c := encoding.GetCompressor("gzip")
+
+	tests := []struct {
+		name                  string
+		compressor            encoding.Compressor
+		input                 mem.BufferSlice
+		maxReceiveMessageSize int
+		want                  mem.BufferSlice
+		compressedSize        int
+		error                 error
+	}{
+		{
+			name:       "Decompresses successfully with sufficient buffer size",
+			compressor: c,
+			input: func() mem.BufferSlice {
```

Comment:
As suggested, change this to []byte and do `mem.BufferSlice` when calling `decompress`